scrypt.go

```diff
@@ -0,0 +1,37 @@
+//go:build !cmd_go_bootstrap
+
+package openssl
+
+// #include "goopenssl.h"
+import "C"
+import (
+	"unsafe"
+)
+
+func Scrypt(password string, salt []byte, N, r, p, maxmem, keylen uint64) ([]byte, error) {
+	cpassword := C.CString(password)
+	defer C.free((unsafe.Pointer)(cpassword))
+
+	csalt := C.CBytes(salt)
+	defer C.free((unsafe.Pointer)(csalt))
+
+	key := C.malloc(C.size_t(keylen))
+	defer C.free(key)
+
+	res := C.go_openssl_EVP_PBE_scrypt(
+		cpassword,
+		C.size_t(len(password)),
+		(*C.uchar)(csalt),
```

Comment:
Same as with `password`, no need to copy `salt` as it is not being modified:

```suggestion
		base(salt),
```